### PR TITLE
Feature: Book entity and service implementation

### PR DESCRIPTION
**Pull request by skfl**
Add implementation of book entity and corresponding service

### CODE REVIEW

File: src/main/java/com/andersentask/bookshop/mappers/BookMapper.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        return Book.builder()
+                .id(dto.getId())
+                .name(dto.getName())
+                .price(dto.getPrice())
+                .status(dto.getStatus())
+                .build();
+    }
+
+    public static List<BookDTO> entityListToDtoList(List<Book> entities) {
+        return entities.stream().map(BookMapper::entityToDto).toList();
```
Comment (DenisBratuh) [inline src/main/java/com/andersentask/bookshop/mappers/BookMapper.java:29]: chaining methods
code convention
and next method

File: src/main/java/com/andersentask/bookshop/services/implementations/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Override
+    public List<BookDTO> getAllBooks() {
+        return BookMapper.entityListToDtoList(bookRepository.findAll());
+    }
+
+    @Override
+    public List<BookDTO> getBooksSortedByName() {
+        List<BookDTO> books = getAllBooks();
+        return books.stream().sorted(Comparator.comparing(BookDTO::getName)).collect(Collectors.toList());
```
Comment (DenisBratuh) [inline src/main/java/com/andersentask/bookshop/services/implementations/BookServiceImpl.java:32]: chaining methods
code convention
and next method

File: src/main/java/com/andersentask/bookshop/entities/Book.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Entity
```
[the remark this answers is not in the document]
Comment (skfl) [reply src/main/java/com/andersentask/bookshop/entities/Book.java:14]: done

File: src/main/java/com/andersentask/bookshop/mappers/BookMapper.java (hunk 1 above)
Comment (skfl) [reply src/main/java/com/andersentask/bookshop/mappers/BookMapper.java]: done

File: src/main/java/com/andersentask/bookshop/services/implementations/BookServiceImpl.java (hunk 2 above)
Comment (skfl) [reply src/main/java/com/andersentask/bookshop/services/implementations/BookServiceImpl.java]: done

**Review by DenisBratuh — APPROVED** [review]

**Pull request merged**